src/plugins/blockresolver/blockresolver.c

```diff
@@ -367,7 +369,7 @@ int elektraBlockresolverSet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned E
 		fin = fopen (data->tmpFile, "r");
 		if (!fin)
 		{
-			ELEKTRA_SET_RESOURCE_ERRORF (parentKey, "Couldn't open %s for reading", data->tmpFile);
+			ELEKTRA_SET_RESOURCE_ERRORF (parentKey, "Couldn't open %s for reading Reason: %s", data->tmpFile, strerror (errno));
```

Comment:
Because I saw it just now: Please make sure to also have guidelines (also written down) of how the error texts should be formatted. E.g. should there be a dot after a sentence? Or a comma and continue "reason" with small letters? Or should it read `"for reading because of '%s'"`? And how to enclose strings so that it is clear where they start and where they end?

Your work now is the best opportunity to make this consistent.

If you see that macros would be helpful please introduce them, see also #2697. Otherwise guidelines will be essential.

Reply:
I added a short guideline. Please take a look if it fits the needs. https://github.com/ElektraInitiative/libelektra/pull/2686/commits/405167cbad8753f33f5bb6fae7eea250d3603a83

> If you see that macros would be helpful please introduce them
Honestly this would make coding in elektra even more frustrating for newcomers because it is even more to know which is not self explanatory but gets rejected in a new PR.
I also think error messages are in general too heterogeneous to put into a macro but if I see something I will add it.